cstrml/market/src/lib.rs

```diff
@@ -252,7 +351,13 @@ decl_module! {
             // 4. Check client has enough currency to pay
             ensure!(T::Currency::can_reserve(&who, amount.clone()), Error::<T>::InsufficientCurrency);
 
-            // 5. Construct storage order
+            // 5. Check if provider pledged
+            ensure!(<PledgeLedgers<T>>::contains_key(&provider), Error::<T>::InsufficientPledge);
+
+            // 6. Check provider has unused pledge
+            ensure!(amount <= Self::pledge_ledgers(&provider).total - Self::pledge_ledgers(&provider).used, Error::<T>::InsufficientPledge);
```

Comment:
no problem. I would use a tmp var. Based on the substrate's doc, I think it would be just one DB Read. substrate have a buffer layer for db's read and write.

Reply:
https://substrate.dev/docs/en/knowledgebase/runtime/fees#parameterizing-over-database-accesses